src/app/account/account-form.tsx

```diff
@@ -0,0 +1,135 @@
+"use client"
+
+import { useCallback, useEffect, useState } from "react"
+import { Database } from "../../types/supabase"
+import {
+  Session,
+  createClientComponentClient,
+} from "@supabase/auth-helpers-nextjs"
+
+export default function AccountForm({ session }: { session: Session | null }) {
+  const supabase = createClientComponentClient<Database>()
+  const [loading, setLoading] = useState(true)
+  const [fullName, setFullName] = useState<string | null>(null)
+  const [username, setUsername] = useState<string | null>(null)
+  const [website, setWebsite] = useState<string | null>(null)
+  const [avatarUrl, setAvatarUrl] = useState<string | null>(null)
+  const user = session?.user
+
+  const getProfile = useCallback(async () => {
+    try {
+      setLoading(true)
+
+      let { data, error, status } = await supabase
+        .from("profiles")
+        .select(`full_name, username, website, avatar_url`)
+        .eq("id", user?.id)
+        .single()
+
+      if (error && status !== 406) {
+        throw error
+      }
+
+      if (data) {
+        setFullName(data.full_name)
+        setUsername(data.username)
+        setWebsite(data.website)
+        setAvatarUrl(data.avatar_url)
+      }
+    } catch (error) {
+      alert("Error loading user data!")
```

Comment:
bug: I always get an error here. Is it expected?

```js
{
    "code": "42703",
    "details": null,
    "hint": null,
    "message": "column \"avatarUrl\" of relation \"profiles\" does not exist"
}
```

Reply:
I think it's because I haven't protected that route, need to do that cuz if u're not logged in it will throw an error

Reply:
this should be solved now that the middleware works